iml-agent/src/daemon_plugins/ostpool.rs

```diff
@@ -0,0 +1,259 @@
+
+            let addlist: HashSet<_> = new.difference(&old).cloned().collect();
+            if !addlist.is_empty() {
+                hm.insert(pool.name.clone(), old.union(&addlist).cloned().collect());
+                changes.push((
+                    OstPoolAction::Grow,
+                    OstPool {
+                        name: pool.name.clone(),
+                        filesystem: fs.clone(),
+                        osts: addlist.iter().cloned().collect(),
+                    },
+                ));
+            }
+
+            let remlist: Vec<String> = old.difference(&new).cloned().collect();
+            if !remlist.is_empty() {
+                changes.push((
+                    OstPoolAction::Shrink,
+                    OstPool {
+                        name: pool.name.clone(),
+                        filesystem: fs.clone(),
+                        osts: remlist,
+                    },
+                ));
+            }
+            hm.insert(pool.name.clone(), pool.osts.clone());
+        } else {
+            hm.insert(pool.name.clone(), pool.osts.clone());
+            changes.push((OstPoolAction::Add, pool.clone()));
+        }
+    }
+
+    // Removed pools
+    for name in rmlist.keys() {
+        let osts = match hm.remove(&name.clone()) {
+            Some(o) => o.clone(),
+            None => vec![],
+        };
+        changes.push((
+            OstPoolAction::Remove,
+            OstPool {
+                name: name.clone(),
+                filesystem: fs.clone(),
+                osts,
+            },
+        ));
+    }
+    Ok(changes)
+}
+
+impl DaemonPlugin for PoolState {
+    fn start_session(
+        &mut self,
+    ) -> Pin<Box<dyn Future<Output = Result<Output, ImlAgentError>> + Send>> {
+        let state = Arc::clone(&self.state);
+
+        async move {
+            let fslist = list_fs()
+                .await
+                .map_err(|_| Ok::<Vec<String>, ImlAgentError>(vec![]))
+                .unwrap();
+
+            let xs = fslist.into_iter().map(|fsname| {
+                let fsmap: HashMap<String, Vec<String>> = HashMap::new();
+                let state = Arc::clone(&state);
+                async move {
+                    state.lock().await.insert(fsname.clone(), fsmap);
+
+                    let xs = pool_list(&fsname)
+                        .await
+                        .map_err(|_| Ok::<Vec<String>, ImlAgentError>(vec![]))
+                        .unwrap()
+                        .into_iter()
+                        .map(|pool| {
+                            let fsname = fsname.clone();
+                            let state = Arc::clone(&state);
+                            async move {
+                                if let Ok(osts) = ost_list(&fsname, &pool)
+                                    .await
+                                    .map_err(|_| Ok::<Vec<String>, ImlAgentError>(vec![]))
```

Comment:
Because of the if let Ok(...) above, this map_err has no effect. This means if ost_list returns an error nothing will happen. Is this what we want to occur?